subsys/mgmt/mcumgr/smp_shell.c

```diff
@@ -98,23 +98,26 @@ size_t smp_shell_rx_bytes(struct smp_shell_data *data, const uint8_t *bytes,
 
 		if (mcumgr_state == SMP_SHELL_MCUMGR_STATE_NONE) {
 			break;
+		} else if (mcumgr_state == SMP_SHELL_MCUMGR_STATE_HEADER &&
+			   !data->buf) {
+			data->buf = net_buf_alloc(data->buf_pool, K_NO_WAIT);
```

Comment:
maybe it would be worth adding log warning if allocation fails.

Reply:
Added LOG_WRN when alloc fails.